qiskit/algorithms/factorizers/shor.py

```diff
@@ -375,9 +392,48 @@ def _calculate_continued_fraction(b: array.array) -> int:
         logger.debug("Numerator:%s \t\t Denominator: %s.", frac.numerator, frac.denominator)
         return frac.denominator
 
-    def _run(self) -> AlgorithmResult:
-        if not self._ret['factors']:
-            logger.debug('Running with N=%s and a=%s.', self._N, self._a)
+    def factor(self,
+               N: int = 15,
+               a: int = 2,
+               ) -> 'ShorResult':
+        """Execute the algorithm.
+
+        The input integer :math:`N` to be factored is expected to be odd and greater than 2.
+        Even though this implementation is general, its capability will be limited by the
+        capacity of the simulator/hardware. Another input integer :math:`a`  can also be supplied,
+        which needs to be a co-prime smaller than :math:`N` .
+
+        Args:
+            N: The integer to be factored, has a min. value of 3.
+            a: Any integer that satisfies 1 < a < N and gcd(a, N) = 1.
+
+        Returns:
+            ShorResult: results of the algorithm.
+
+        Raises:
+            ValueError: Invalid input
+
+        """
+        validate_min('N', N, 3)
+        validate_min('a', a, 2)
+
+        # check the input integer
+        if N < 1 or N % 2 == 0:
+            raise ValueError('The input needs to be an odd integer greater than 1.')
+
+        if a >= N or math.gcd(a, N) != 1:
+            raise ValueError('The integer a needs to satisfy a < N and gcd(a, N) = 1.')
+
+        self._ret = ShorResult()
```

Comment:
Can this be a local variable only instead of a class member?

Reply:
If you are referring to N, it isn't a class member but a function parameter. The original developer used the parameter 'N' uppercase.

Reply:
I meant `self._ret` 🙂

Reply:
Ahh ok, changed.